esp-alloc/src/lib.rs

```diff
@@ -61,35 +67,81 @@ use core::{
 };
 
 use critical_section::Mutex;
+use enumset::{EnumSet, EnumSetType};
 use linked_list_allocator::Heap;
 
+/// The global allocator instance
+#[global_allocator]
+pub static INSTANCE: EspHeap = EspHeap::empty();
+
+const NON_REGION: Option<HeapRegion> = None;
+
+#[derive(EnumSetType)]
+/// Describes the properties of a memory region
+pub enum MemoryCapability {
```

Comment:
I don't really have a better suggestion, but this name feels a bit odd to me. Something like `MemoryLocation` maybe would be okay, but then that also could be conflated with memory addressing and stuff.

Reply:
Okay if there is some precedent here for the naming at least that's a bit better, thanks.